### PR TITLE
Enhancement:  Only look back to FirstValid for pending transactions that were found in the pool

### PR DESCRIPTION
**Pull request by barnjamin**
## Summary

Addresses #3967 to provide a much faster response time when looking for pending transactions.

The current behavior is to look in the pending pool for the transaction ID, if found set a flag and speculatively set the txn var to be returned but continue looking through the last ~1k blocks for the same transaction.

This change uses the information we found in the pending pool (if found) to restrict the historic rounds we search to only those where it could _possibly_ be found.

Since this endpoint is most commonly used to track confirmed transactions, the `FirstValid` round is likely to be recent. So for an unconfirmed transaction that is found in the pool, we don't waste time looking back 1k rounds for it

## Test Plan

Copied bin into sandbox and curl'd it

### CODE REVIEW

File: node/node.go
Comment (jasonpaulos) [inline node/node.go:661]: As long as `tx.Txn.FirstValid` is guaranteed to be before `tx.Txn.LastValid` at this point (which I expect it must be, if the txn is allowed in the pending pool), this looks correct to me

**Review by winder — APPROVED** [review]

**Review by jasonpaulos — APPROVED** [review]

**Comment by barnjamin** [comment]
@algorandskiy I don't have merge rights, can you please take a look at this and approve/merge if it looks reasonable?

**Pull request merged**
